### PR TITLE
ENH: Updated CLI interface

### PR DESCRIPTION
**Pull request by mmarineBlueQuartz**
## Changes:

* Renamed PipelineRunner to NX-cli.
* Restructured parsing for parameters.
  * Changed syntax for preflight and execute.
  * Added syntax for help.
* Renamed namespace from PipelineRunner to CLI

## Naming Conventions

Naming of variables should descriptive where needed. Loop Control Variables can use `i` if warranted. Most of these conventions are enforced through the clang-tidy and clang-format configuration files. See the file `complex/docs/Code_Style_Guide.md` for a more in depth explanation.

## Filter Checklist

The help file `complex/docs/Porting_Filters.md` has documentation to help you port or write new filters. At the top is a nice checklist of items that should be noted when porting a filter.

## Unit Testing

The idea of unit testing is to test the filter for proper execution and error handling. How many variations on a unit test each filter needs is entirely dependent on what the filter is doing. Generally, the variations can fall into a few categories:

- [x] 1 Unit test to test output from the filter against known exemplar set of data
- [x] 1 Unit test to test invalid input code paths that are specific to a filter. Don't test that a DataPath does not exist since that test is already performed as part of the SelectDataArrayAction.

## Code Cleanup
- [x] No commented out code (rare exceptions to this is allowed..)
- [x] No API changes were made (or the changes have been approved)
- [x] No major design changes were made (or the changes have been approved)
- [x] Added test (or behavior not changed)
- [x] Updated API documentation (or API not changed)
- [x] Added license to new files (if any)
- [x] Added example pipelines that use the filter
- [x] Classes and methods are properly documented

### CODE REVIEW

File: src/DREAM3DNX-cli/src/DREAM3DNX_cli.cpp [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    args.emplace_back(ArgumentType::Preflight, argStr);
+  }
+
+  if(argc > index)
+  {
+    std::string ss = "Too many arguments to parse";
+    return complex::MakeErrorResult<CliArguments>(k_FailedParsingArguments, ss);
+  }
+
+  return {args};
```
Comment (joeykleingers) [inline src/DREAM3DNX-cli/src/DREAM3DNX_cli.cpp]: Unless I'm missing it, this algorithm should have a default case that handles when the argument is unrecognized/unexpected.  For example, if argc = 2 and argv = [executable_path, "--blah"], then this code will go all the way through without setting args to anything or erroring out, and then will return empty args.

File: src/DREAM3DNX-cli/src/DREAM3DNX_cli.cpp [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+  std::cout << fmt::format("Preflighting pipeline at path: '{}'\n", pipelinePath) << std::endl;
+
+  Pipeline pipeline = loadPipelineResult.value();
+  return PreflightPipeline(pipeline);
+}
+
+void DisplayDefaultHelp()
+{
+  std::cout << "Options:\n";
+  std::cout << fmt::format("\t {} <filepath>\t", k_ExecuteParamLong) << "\t Execute the pipeline at the taget filepath\n";
```
Comment (joeykleingers) [inline src/DREAM3DNX-cli/src/DREAM3DNX_cli.cpp]:
```suggestion
  std::cout << fmt::format("\t {} <filepath>\t", k_ExecuteParamLong) << "\t Execute the pipeline at the target filepath\n";
```

File: src/DREAM3DNX-cli/src/DREAM3DNX_cli.cpp [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+  std::cout << fmt::format("\t {} <filepath>\t", k_ExecuteParamShort) << "\t Execute the pipeline at the taget filepath\n";
```
Comment (joeykleingers) [inline src/DREAM3DNX-cli/src/DREAM3DNX_cli.cpp]:
```suggestion
  std::cout << fmt::format("\t {} <filepath>\t", k_ExecuteParamShort) << "\t Execute the pipeline at the target filepath\n";
```

File: src/DREAM3DNX-cli/src/DREAM3DNX_cli.cpp [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+  std::cout << fmt::format("\t {} <filepath>\t", k_PreflightParamLong) << "\t Preflight the pipeline at the taget filepath" << std::endl;
```
Comment (joeykleingers) [inline src/DREAM3DNX-cli/src/DREAM3DNX_cli.cpp]:
```suggestion
  std::cout << fmt::format("\t {} <filepath>\t", k_PreflightParamLong) << "\t Preflight the pipeline at the target filepath" << std::endl;
```

File: src/DREAM3DNX-cli/src/DREAM3DNX_cli.cpp [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+  std::cout << fmt::format("\t {} <filepath>\t", k_PreflightParamShort) << "\t Preflight the pipeline at the taget filepath" << std::endl;
```
Comment (joeykleingers) [inline src/DREAM3DNX-cli/src/DREAM3DNX_cli.cpp]:
```suggestion
  std::cout << fmt::format("\t {} <filepath>\t", k_PreflightParamShort) << "\t Preflight the pipeline at the target filepath" << std::endl;
```

**Review by joeykleingers — CHANGES_REQUESTED** [review]

**Comment by imikejackson** [comment]
@mmarineBlueQuartz I would also like to have the option "--logfile" so that all output can be redirected to a file.

**Comment by imikejackson** [comment]
in the context of the flow of events, if possible, I would recommend that the arguments are all figured out, then the plugins are loaded and the pipeline executed. 

Take a look at my changes/additions and see what you think

**Review by imikejackson — APPROVED** [review]

**Review by imikejackson — APPROVED** [review]

**Review by imikejackson — APPROVED** [review]

**Pull request merged**